corporate_partner_access/models/partner_access.py

```diff
@@ -0,0 +1,84 @@
+"""Models for managing course catalogs and access for corporate partners."""
+
+
+from django.contrib.auth import get_user_model
+from django.db import models
+
+from corporate_partner_access.edxapp_wrapper.course_module import course_overview
+
+
+class CorporatePartner(models.Model):
+    """Company that manages one or more course catalogs."""
+
+    id = models.AutoField(primary_key=True)
+    code = models.CharField(max_length=255, unique=True)
+    name = models.CharField(max_length=255)
+    logo = models.ImageField(upload_to="partner_logos/", blank=True, null=True)
+    homepage_url = models.URLField(blank=True, null=True)
+
+    class Meta:
+        verbose_name = "Corporate Partner"
+        verbose_name_plural = "Corporate Partners"
+        ordering = ["name"]
+
+    def __str__(self):
+        return f"<CorporatePartner: {self.name} (Code: {self.code})>"
+
+
+class CorporatePartnerCatalogEmailRegex(models.Model):
+    """Regex pattern for validating emails for a partner's catalog."""
+
+    id = models.AutoField(primary_key=True)
+    catalog = models.ForeignKey(
+        "CorporatePartnerCatalog",
+        on_delete=models.CASCADE,
+        related_name="email_regexes",
+    )
+    regex = models.CharField(
```

Comment:
2 Comments, 1st: We should add a clean to validate that the regex works well, I mean is a valid regex. 2nd: Change the max_length to 500 just in case we want to use a regex more complex.